src/transformers/training_args.py

```diff
@@ -1510,6 +1517,20 @@ class TrainingArguments:
         },
     )
 
+    predict_with_generate: bool = field(
+        default=False, metadata={"help": "Whether to use generate to calculate generative metrics (ROUGE, BLEU)."}
+    )
+    generation_config: Optional[GenerationConfig] = field(
+        default=None,
+        metadata={
+            "help": (
+                "The GenerationConfig that will be used during prediction. Args from this config ",
+                "will have higher priority than model's generation config. Anything not set by this config ",
+                "will fallback to `model.generation_config`.",
+            )
+        },
+    )
```

Comment:
I think we simplify a bit things if we also add a `generation_kwargs` as this is incompatible with `generation_config` + I don't think we want to merge both arguments into one. WDYT @muellerzr ?

Reply:
Hmm, maybe we can then allow users to pass `generation_config` as a dict also, then we can make a Config object of it ourselves. I see that `TrainerSeq2Seq` args also uses a `config` arg, so I thought we could later merge `seq2seq` args with `trainerArgs`

Reply:
I would be better I think ! This way, we won't need to have  **gen_kwargs in evaluate and predict function. cc @muellerzr @gante

Reply:
oke, now we can accept a dict or a config object in training args